tests/integration_tests/tests/agentless_tests/test_plugins.py

```diff
@@ -83,28 +93,86 @@ def test_delete_plugin_not_found(self):
+
+
+class TestPluginsSystemState(AgentlessTestCase):
+    def test_installing_corrupted_plugin_doesnt_affect_system_integrity(self):
+        self._upload_plugin_and_assert_values(TEST_PACKAGE_NAME,
+                                              TEST_PACKAGE_VERSION,
+                                              plugins_count=0,
+                                              # Wagon has a bug that leaves 1
+                                              # wagon file at /tmp
+                                              wagon_files_count=1,
+                                              corrupt_plugin=True)
+        plugin = self._upload_plugin_and_assert_values(TEST_PACKAGE_NAME,
+                                                       TEST_PACKAGE_VERSION,
+                                                       plugins_count=1,
+                                                       # Same reason as above,
+                                                       # 2 for wagon's bug
+                                                       # (accumulated),
+                                                       # 1 for the file at
+                                                       # ".../resources/"
+                                                       wagon_files_count=3)
+        plugin2 = self._upload_plugin_and_assert_values(TEST_PACKAGE2_NAME,
+                                                        TEST_PACKAGE2_VERSION,
+                                                        plugins_count=2,
+                                                        # 1 for wagon's bug
+                                                        # and one at
+                                                        # ".../resources/".
+                                                        # Note: this uses a
+                                                        # different plugin
+                                                        # than above.
+                                                        wagon_files_count=2)
+        self._uninstall_plugin_and_assert_values(plugin, 1)
+        self._uninstall_plugin_and_assert_values(plugin2, 0)
+
+    def _upload_plugin_and_assert_values(self,
+                                         package_name,
+                                         package_version,
+                                         plugins_count,
+                                         wagon_files_count,
+                                         corrupt_plugin=False):
+        _plugin = test_utils.upload_mock_plugin(
+            package_name,
+            package_version,
+            corrupt_plugin=corrupt_plugin)
+        wagon_file = _plugin.archive_name[len(INSTALLING_PREFIX):]
+
+        ex = self._get_latest_execution('install_plugin')
+        if corrupt_plugin:
+            with self.assertRaisesRegexp(
+                    RuntimeError, 'Workflow execution failed: .*'):
+                self.wait_for_execution_to_end(ex)
```

Comment:
won't u need always to wait for the execution to finish?

Reply:
This code block always waits for the `install_plugin` execution. If that's what you mean

Reply:
So why do we need to add here the wait for execution?